### ISSUE TITLE
Submitting via Ajax from same server / another domain [need help]

### ISSUE THREAD

**Issue by crazyx13th** [craftcms/guest-entries#24]
Hey there,

I'm trying to add entries from my "api" to the Craft-CMS-3 system. But the ajax call isn't working.

```
request
	.post({
		url: 'http://staging.cms.xxxxx.com',
		headers: {
			'Accept': 'application/json',
			'content-type': 'application/json'
		}
	})
	.form({
		action: 'guest-entries/save',
		sectionId: 11,
		title: req.body.title
	})
	.on('response', function (response: any) {
		res.end(JSON.stringify(response));
	});
```
but there isn't any entry in my cms and the response isn't a json

```
{
	"statusCode": 302,
	"headers": {
		"server": "nginx",
		"date": "Tue, 01 Aug 2017 19:08:23 GMT",
		"content-type": "text/html; charset=iso-8859-1",
		"content-length": "291",
		"connection": "close",
		"location": "https://staging.cms.xxxxx.com/"
	},
	"request": {
		"uri": {
			"protocol": "http:",
			"slashes": true,
			"auth": null,
			"host": "staging.cms.xxxxx.com",
			"port": 80,
			"hostname": "staging.cms.xxxxx.com",
			"hash": null,
			"search": null,
			"query": null,
			"pathname": "/",
			"path": "/",
			"href": "http://staging.cms.xxxxx.com/"
		},
		"method": "POST",
		"headers": {
			"Accept": "application/json",
			"content-type": "application/x-www-form-urlencoded",
			"content-length": 73
		}
	}
}
```
fyi: I can't change the header "content-type"

thx!!

Greetings crazyx13th

**Comment by brandonkelly** [comment]
You need to set the `Accept` header to `application/json`.

```js
// with XMLHttpRequest:
xhr.setRequestHeader('Accept', 'application/json');

// with jQuery:
$.post({
  dataType: 'json',
  // ...
});
```

**Comment by crazyx13th** [comment]
ehm yeah, this is what I did, look at my headers, thx

**Comment by brandonkelly** [comment]
Sorry, missed that.  You also need to either set the URL to `.../actions/guest-entries/save`, or include an `action` body param with the value `guest-entries/save`.

**Comment by crazyx13th** [comment]
i had an **action** in my request, just look at my params ;-)
I tried now a lot with frontend jquery **ajax**, but at the end this was a first milestone:
https://stackoverflow.com/a/41781597/8270748
But than I can't call from an external domain, right? because I need the craft token, right?
thx!

**Comment by brandonkelly** [comment]
Sorry not familiar with the JS methods you’re using. What is `request`?

**Comment by crazyx13th** [comment]
this was was my first attempt to save data into craft from the backend/api

what i'm wrinting in my last post I try it (for better debugging) at the frontend with jQuery like you say before. But the request get an error-400 without an csrfToken.

I allways need an csrfToken, right? than it will not work to save data from a another domain (same server), right?

thx

**Comment by brandonkelly** [comment]
> I allways need an csrfToken, right? than it will not work to save data from a another domain (same server), right?

Correct, since the CSRF token has to be sent in part as a cookie. You could do it from a different subdomain though, if you set the `defaultCookieDomain` config setting to `.domain.com`.

That said, maybe CSRF isn’t really necessary for Guest Entries, since it’s for anonymous users, so wouldn’t be much of a target for CSRF attacks. Maybe I’ll add a config setting for it…

**Comment by brandonkelly** [comment]
Just released Guest Entries 2.1 with a “Enable CSRF Validation?” setting, which you can disable, and will only affect `guest-entries/save` requests.

**Comment by crazyx13th** [comment]
awesome! thx 👍
